### PR TITLE
[CLNP-2532] fix: html xss

### PR DESCRIPTION
**Pull request by bang9**
- replace `dangerouslySetInnerHTML` with `HTML tag` to `RegexText` component
- support markdown URL format(`[text](url)`)
- support markdown bold format(`**bold**`)

ticket: [CLNP-2532]

[CLNP-2532]: https://sendbird.atlassian.net/browse/CLNP-2532?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -31,18 +31,19 @@
...
     "polished": "^2.3.1",
     "react-code-blocks": "^0.1.0",
     "react-popper-tooltip": "^4.4.2",
-    "styled-components": "^5.3.11",
-    "@tanstack/react-query": "^5.17.19"
+    "styled-components": "^5.3.11"
   },
   "devDependencies": {
+    "@tanstack/eslint-plugin-query": "^5.17.22",
+    "@types/dompurify": "^3.0.5",
```
Comment (AhyoungRyu) [inline package.json:43]: 👍

**Review by AhyoungRyu — APPROVED** [review]
I tested and it works well. Thanks!

File: src/components/ParsedBotMessageBody.tsx [hunk 2]
```diff
@@ -90,3 +123,45 @@ export default function ParsedBotMessageBody(props: Props) {
...
+  children: string;
+  patterns: RegexTextPattern[];
+}) => {
+  if (patterns.length === 0 || typeof children !== 'string') {
+    return <>{children}</>;
+  }
+
+  const convertedNodes: Array<string | ReactNode> = [children];
+  patterns.forEach(({ regex, replacer }) => {
+    const node = convertedNodes.concat();
```
Comment (chohongm) [inline src/components/ParsedBotMessageBody.tsx:159]: what is point of concat without arg?

File: src/components/ParsedBotMessageBody.tsx [hunk 3]
```diff
@@ -90,3 +123,45 @@ export default function ParsedBotMessageBody(props: Props) {
...
+    let offset = 0;
+    node.forEach((text, index) => {
+      if (typeof text === 'string' && text) {
+        const children = replaceWithRegex(text, regex, replacer);
+
+        if (children.length > 1) {
+          convertedNodes.splice(index + offset, 1, ...children);
```
Comment (chohongm) [inline src/components/ParsedBotMessageBody.tsx:166]: children is a string right? why we need spread operator here?

File: src/components/ParsedBotMessageBody.tsx (hunk 2 above)
Comment (bang9) [reply src/components/ParsedBotMessageBody.tsx:159]: It just copies the array
it's the same as `const node = [...convertedNodes];`

File: src/components/ParsedBotMessageBody.tsx (hunk 3 above)
Comment (bang9) [reply src/components/ParsedBotMessageBody.tsx:166]: children is a `Array<string | ReactNode>` type

**Pull request merged**